lib/core/test/unit/Cardano/Wallet/Primitive/SlottingSpec.hs

```diff
@@ -111,6 +145,72 @@ spec = do
                 run (endTimeOfEpoch e)
                     === run (slotToUTCTime =<< firstSlotInEpoch (e + 1))
 
+        let setupTestLogging = do
+                tvar <- newTVarIO []
+                let tr = traceInTVarIO tvar
+                return (tr, tvar)
+        before setupTestLogging $
+            describe "TimeInterpreter conversions beyone the safe zone" $ do
+
+            startTime <- runIO $ StartTime <$> getCurrentTime
+            let failingQry = slotToUTCTime (SlotNo 100000)
+
+            it "normally fails and logs failures as Notice" $ \(tr, tvar) -> do
+                let ti = mkTimeInterpreter tr startTime (pure forkInterpreter)
+                res <- runExceptT $ interpretQuery ti failingQry
+                logs <- readTVarIO tvar
```

Comment:
Extracting the common parts with a test helper was more tricky than I expected due to the different return-types, so leaving it like this with some duplication.

Reply:
I changed it to use `Test.Utils.Trace.captureLogging`.